src/librustc_ast_lowering/pat.rs

```diff
@@ -102,10 +103,34 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             // Note that unlike for slice patterns,
             // where `xs @ ..` is a legal sub-slice pattern,
             // it is not a legal sub-tuple pattern.
-            if pat.is_rest() {
-                rest = Some((idx, pat.span));
-                break;
+            match pat.kind {
+                // Found a sub-tuple rest pattern
+                PatKind::Rest => {
+                    rest = Some((idx, pat.span));
```

Comment:
Shouldn't we also have this line in the other arm? That way we _won't_ emit the mismatched pattern size error, which would be solved if the pattern is changed. We would _still_ emit errors if `ident` is being used inside the arm, but that should be ok.

Reply:
Looking into this.

Reply:
This is correct Esteban.  I will push a new commit with this change.

Reply:
https://github.com/rust-lang/rust/pull/72677/commits/e73ac086c54084dced1916d062267be4058b257c